### PR TITLE
video

### PR DESCRIPTION
**Pull request by gintama91**
### Description

Video

### Image(if needed, helps for a faster review)

Screenshot_20230707_151603

### Checklist

- [ ] Run tests locally
- [ ] Run linter(check for linter errors)

### CODE REVIEW

File: lib/scarpe/wv/video.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  class WebviewVideo < Scarpe::WebviewWidget
+    def initialize(properties)
+      @url = properties[:url]
+      super
+    end
+
+    def element
+      HTML.render do |h|
+        h.video(id: html_id, style: style, controls: true) do
+          h.source(src: @url, type: "video/mp4")
```
Comment (noahgibbs) [inline lib/scarpe/wv/video.rb:13]: If we only support mp4, we should check that in the Shoes-side widget and give an error if they try to use something without an mp4-ish extension.

At some point we should figure out which formats are possible for which webview platforms.

Comment (gintama91) [reply lib/scarpe/wv/video.rb:13]: hm is dng smthng like ok?

```
    def get_type
      file_extension = File.extname(@url).downcase
      case file_extension
      when '.mp4'
        'video/mp4'
      when '.webp'
        'video/webp'
      else
        'video/mp4' # hm default?
      end
    end
    ```

Comment (gintama91) [reply lib/scarpe/wv/video.rb:13]: or maybe maybe we can use mime this but i think its overkill

Comment (gintama91) [reply lib/scarpe/wv/video.rb:13]: oo or this? https://docs.ruby-lang.org/en/2.2.0/WEBrick/HTTPUtils.html

Comment (noahgibbs) [reply lib/scarpe/wv/video.rb:13]: I think the first bit with get_type will be fantastic. There's plenty of time to overengineer later, once it's clear where to do it :-)

**Comment by gintama91** [comment]
changed  to support more formats

**Review by noahgibbs — APPROVED** [review]

**Pull request merged**